pkg/didcomm/transport/http/outbound.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"bytes"
+	"crypto/tls"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+
+	tlsCertPool "github.com/hyperledger/aries-framework-go/pkg/didcomm/transport/http/tls"
+	"github.com/pkg/errors"
+)
+
+const commContentType = "application/didcomm-envelope-enc"
+
+// OutboundCommHTTP is the HTTP transport implementation of CommTransport
+// it embeds an http.Server and has an http.Client instance
+type OutboundCommHTTP struct {
+	client *http.Client
+	cfg    *OutboundCommConfig
+}
+
+// OutboundCommConfig are the HTTP client config used by OutboundCommHTTP
+type OutboundCommConfig struct {
+	Timeout      time.Duration
+	CACertsPaths string
+	caCertPool   tlsCertPool.CertPool
+}
+
+// NewOutboundCommFromConfig creates a new instance of CommHTTP to Post requests to other Agents
+// To get a valid instance of OutboundCommHTTP, a cfg must be set
+// if config is nil, this function will throw an error
+func NewOutboundCommFromConfig(cfg *OutboundCommConfig) (*OutboundCommHTTP, error) {
+	var cl *http.Client
+	if cfg != nil {
+		var err error
+		cl, err = newHTTPClient(cfg)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, errors.New("config is empty, cannot create new HTTP transport")
+	}
+
+	cm := &OutboundCommHTTP{
+		client: cl,
+		cfg:    cfg,
+	}
+	return cm, nil
+}
+
+// NewOutboundCommFromClient creates a new instance of CommHTTP to Post requests to other Agents
+// To get a valid instance of OutboundCommHTTP, a client must be set
+// if client is nil, this function will throw an error
+func NewOutboundCommFromClient(client *http.Client) (*OutboundCommHTTP, error) {
+	if client == nil {
+		return nil, errors.New("client is empty, cannot create new HTTP transport")
+	}
+
+	cm := &OutboundCommHTTP{
+		client: client,
+	}
+	return cm, nil
+}
+
+// Send sends a2a exchange data via HTTP (client side)
+func (cs *OutboundCommHTTP) Send(data string, url string) (string, error) {
+	resp, err := cs.client.Post(url, commContentType, bytes.NewBuffer([]byte(data)))
+	if err != nil {
+		log.Printf("HTTP Transport - Error posting did envelope to agent at [%s]: %v", url, err)
```

Comment:
This should be error logger. May be we can add a TODO here to change once we have a logging framework in the project.

Reply:
the intent is to replace this by a logger framework (and add possibly a few more logs around the code). This can be done a separate PR